### PR TITLE
provider/kubernetes: Liveness & readiness probes

### PR DESCRIPTION
**Pull request by lwander**
This is the equivalent of a health checking mechanism for Kubernetes. More information is here: http://kubernetes.io/v1.1/docs/user-guide/liveness/README.html

@duftler

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy [hunk 1]
```diff
@@ -179,6 +184,9 @@ class KubernetesApiAdaptor {
       return port
     }
 
+    containerDescription.livenessProbe = container?.livenessProbe ? fromProbe(container.livenessProbe) : null
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy]: Could just have `fromProbe()` return null if the passed arg is null. But I can live with it this way if you'd rather not.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy [hunk 2]
```diff
@@ -200,4 +208,39 @@ class KubernetesApiAdaptor {
...
+    def kubernetesTcpSocketAction = new KubernetesTcpSocketAction()
+    kubernetesTcpSocketAction.port = tcpSocket.port.intVal
+    return kubernetesTcpSocketAction
+  }
+
+  static KubernetesHttpGetAction fromHttpGetAction(HTTPGetAction httpGet) {
+    def kubernetesHttpGetAction = new KubernetesHttpGetAction()
+    kubernetesHttpGetAction.host = httpGet.host
+    kubernetesHttpGetAction.path = httpGet.path
+    kubernetesHttpGetAction.port = httpGet.port.intVal
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy]: Can `port` ever be null?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy [hunk 3]
```diff
@@ -145,6 +150,76 @@ class DeployKubernetesAtomicOperation implements AtomicOperation<DeploymentResul
         }
       }
 
+      [liveness: container.livenessProbe, readiness: container.readinessProbe].each { k, v ->
+        def probe = v
+        if (probe) {
+          if (k == 'liveness') {
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy]: Pretty sure groovy's `switch` statements can operate on `String`s.

**Comment by duftler** [comment]
LGTM after responding to comments.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy (hunk 1 above)
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy]: I was debating doing it like that, I'll switch it over.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy (hunk 2 above)
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiAdaptor.groovy]: Ah you're right, it shouldn't ever be.

**Pull request merged**